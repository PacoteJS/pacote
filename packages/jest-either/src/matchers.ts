export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
