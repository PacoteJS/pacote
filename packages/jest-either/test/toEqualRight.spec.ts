import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src'

expect.extend(matchers)

describe('.toEqualRight()', () => {
  test('passes when object is a right', () => {
    expect(right({ a: { b: 'right' } })).toEqualRight({ a: { b: 'right' } })
  })

  test('fails when object is not equal', () => {
    expect(() =>
      expect(right({ a: { b: 'right' } })).toEqualRight('wrong')
    ).toThrowErrorMatchingSnapshot()
  })

  test('fails when object is a left', () => {
    expect(() =>
      expect(left('left')).toEqualRight('left')
    ).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toEqualRight()', () => {
  test('passes when object is a left', () => {
    expect(left('left')).not.toEqualRight('left')
  })

  test('passes when object is not equal', () => {
    expect(right({ a: { b: 'right' } })).not.toEqualRight('wrong')
  })

  test('fails when object is a right', () => {
    expect(() =>
      expect(right('right')).not.toEqualRight('right')
    ).toThrowErrorMatchingSnapshot()
  })
})
