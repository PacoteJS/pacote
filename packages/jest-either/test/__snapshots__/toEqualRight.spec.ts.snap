// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toEqualRight() fails when object is a right 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toEqualRight([22m[32mexpectedRight[39m[2m)[22m

Expected Either not to equal right:
  [31m\\"same\\"[39m

Received right:
  [31m\\"same\\"[39m"
`;

exports[`.toEqualRight() fails when object is a left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toEqualRight([22m[32mexpectedRight[39m[2m)[22m

Expected Either to equal right:
  [31m\\"same\\"[39m

Received left."
`;

exports[`.toEqualRight() fails when object is not equal 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toEqualRight([22m[32mexpectedRight[39m[2m)[22m

Expected Either to equal right:
  [31m\\"different\\"[39m

Received right:
  [31m{\\"test\\": \\"ok\\"}[39m"
`;
